Test/AppVeyorWaitForBuildJobs.Tests.ps1

```diff
@@ -0,0 +1,474 @@
+                    continue
+                }
+
+                if( -not ($job | Get-Member 'finished') )
+                {
+                    Write-Timing ('[{0}]  Adding "finished" property.' -f $job.name)
+                    Add-Member -InputObject $job -Name 'finished' -MemberType 'NoteProperty' -Value ((Get-Date).ToString('s'))
+                }
+
+                if( $job.status -ne $job.finalStatus )
+                {
+                    Write-Timing ('[{0}]  Setting final status to "{1}".' -f $job.name,$job.finalStatus)
+                    $job.status = $job.finalStatus
+                }
+            }
+            return $project
+        }.GetNewClosure()
+
+    $script:failed = $false
+    try
+    {
+        $parameter = @{}
+        $task = $context.Configuration['Build'][0]
+        if( $task -isnot [string] )
+        {
+            $parameter = $task['AppVeyorWaitForBuildJobs']
+        }
+        
+        Invoke-WhiskeyTask -TaskContext $context -Name 'AppVeyorWaitForBuildJobs' -Parameter $parameter
+    }
+    catch
+    {
+        $script:failed = $true
+        Write-Error -ErrorRecord $_
+    }
+    finally
+    {
+        Remove-Variable -Name 'CheckNum' -Scope 'Global'
+    }
+}
+
+Describe 'AppVeyorWaitForBuildJobs.when there is only one job' {
+    It 'should immediately finish' {
+        Init
+        GivenRunBy -AppVeyor
+        GivenSecret 'fubarsnafu' -WithID 'AppVeyor'
+        GivenWhiskeyYml @'
+Build:
+- AppVeyorWaitForBuildJobs:
+    ApiKeyID: AppVeyor
+    CheckInterval: 00:00:00.1
+'@
+        GivenJob -Current 
+        WhenRunningTask
+        ThenSucceeds
+        ThenCheckedStatus -Times 1 -Exactly
+    }
+}
+
+Describe 'AppVeyorWaitForBuildJobs.when there are two jobs' {
+    It 'should wait for second job to finish' {
+        Init
+        GivenRunBy -AppVeyor
+        GivenSecret 'fubarsnafu' -WithID 'AppVeyor'
+        GivenWhiskeyYml @'
+Build:
+- AppVeyorWaitForBuildJobs:
+    ApiKeyID: AppVeyor
+    CheckInterval: 00:00:00.1
+'@
+        GivenJob -WithStatus 'running' -ThatFinishesAtCheck 2 -WithFinalStatus 'success'
+        GivenJob -Current 
+        WhenRunningTask
+        ThenSucceeds
+        ThenCheckedStatus -Times 2 
+    }
+}
+
+Describe 'AppVeyorWaitForBuildJobs.when there are three jobs and one job takes awhile' {
+    It 'should wait for all jobs to finish' {
+        Init
+        GivenRunBy -AppVeyor
+        GivenSecret 'fubarsnafu' -WithID 'AppVeyor'
+        GivenWhiskeyYml @'
+Build:
+- AppVeyorWaitForBuildJobs:
+    ApiKeyID: AppVeyor
+    CheckInterval: 00:00:00.1
+    ReportInterval: 00:00:00.1
+'@
+        GivenJob -WithStatus 'running' -ThatFinishesAtCheck 2 -WithFinalStatus 'success'
+        GivenJob -Current 
+        GivenJob -WithStatus 'running' -ThatFinishesAtCheck 10 -WithFinalStatus 'success'
+        WhenRunningTask
+        ThenSucceeds
+        ThenCheckedStatus -Times 10 

```

Comment:
Also add a test that ensures the task is sleeping for the specified interval? It looks like this just ensures the task iterates a number of times while checking jobs.